src/sagemaker_xgboost_container/algorithm_mode/train.py

```diff
@@ -265,15 +282,48 @@ def train_job(train_cfg, train_dmatrix, val_dmatrix, train_val_dmatrix, model_di
                                     evals=watchlist, feval=configured_feval, evals_result=evals_result,
                                     callbacks=callbacks, xgb_model=xgb_model, verbose_eval=False)
                 bst.append(booster)
+
+                # store predictions on validation fold
+                data_to_store[val_idx, cv_repeat_idx[val_idx], truth_idx] = cv_val_dmatrix.get_label()
+                if classification_problem:
+                    probabilities = booster.predict(cv_val_dmatrix)
+                    if probabilities.ndim == 2:
+                        # multi - class classification setting
+                        pred_labels = probabilities.argmax(axis=-1)
+                        probabilities = probabilities[np.arange(len(probabilities)), pred_labels]
+                    else:
+                        pred_labels = 1*(probabilities > 0.5)
+                    data_to_store[val_idx, cv_repeat_idx[val_idx], prob_idx] = probabilities
+                    data_to_store[val_idx, cv_repeat_idx[val_idx], pred_idx] = pred_labels
+                else:
+                    data_to_store[val_idx, cv_repeat_idx[val_idx], pred_idx] = booster.predict(cv_val_dmatrix)
+
+                cv_repeat_idx[val_idx] += 1
                 evals_results.append(evals_result)
 
                 if len(bst) % kfold == 0:
                     logging.info("The metrics of round {} cross validation".format(int(len(bst) / kfold)))
                     print_cv_metric(num_round, evals_results[-kfold:])
 
+            if classification_problem:
```

Comment:
similarly, for the "closing/writing" step, can we also factor this out into a helper function like _train_job_save_postprocessing_data?

Reply:
Refactored to a dedicated class